speckle_toolbox/esri/toolboxes/speckle/ui/project_vars.py

```diff
@@ -0,0 +1,253 @@
+            
+            if self.active_map is not None and isinstance(self.active_map, Map): # if project loaded
+                for layer in self.active_map.listLayers(): 
+                    #print(layer)
+                    if layer.isFeatureLayer or layer.isRasterLayer: self.all_layers.append(layer) #type: 'arcpy._mp.Layer'
+            self.stream_file_path: str = aprx.filePath.replace("aprx","gdb") + "\\speckle_streams.txt"
+
+            if os.path.exists(self.stream_file_path): 
+                try: 
+                    f = open(self.stream_file_path, "r")
+                    content = f.read()
+                    self.saved_streams = self.getProjectStreams(content)
+                    f.close()
+                except: pass
+                
+            else: 
+                f = open(self.stream_file_path, "w")
+                content = ""
+                f.write(content)
+                f.close()
+        except: self.project = None; print("Project not found")
+        self.instances.append(self)
+
+    def getProjectStreams(self, content: str = None):
+        print("get proj streams")
+        if not content: 
+            content = self.stream_file_path
+            try: 
+                f = open(self.stream_file_path, "r")
+                content = f.read()
+                f.close()
+            except: pass
+
+        ######### need to check whether saved streams are available (account reachable)
+        if content:
+            streamsTuples = []
+            for i, url in enumerate(content.split(",")):
+
+                streamExists = 0
+                index = 0
+                try:
+                    sw = StreamWrapper(url)
+                    stream = self.tryGetStream(sw)
+
+                    for st in streamsTuples: 
+                        if isinstance(stream, Stream) and st[0].stream_id == stream.id: 
+                            streamExists = 1; 
+                            break 
+                        index += 1
+                    if streamExists == 1: del streamsTuples[index]
+                    streamsTuples.insert(0,(sw, stream))
+
+                except SpeckleException as e:
+                    arcpy.AddMessage(str(e.args[0]))
+            return streamsTuples
+        else: return []
+
+    def tryGetStream (self,sw: StreamWrapper) -> Stream:
+        #print("Try get streams")
+
+        steamId = sw.stream_id
+        try: steamId = sw.stream_id.split("/streams/")[1].split("/")[0] 
+        except: pass
+
+        client = sw.get_client()
+        stream = client.stream.get(steamId)
+        if isinstance(stream, GraphQLException):
+            raise SpeckleException(stream.errors[0]['message'])
+        return stream
+
+
+class toolboxInputsClass:
+    #def __init__(self):
+    print("CREATING UI inputs first time________")
+    # self.instances.append(self)
+    instances = []
+    lat: float = 0.0
+    lon: float = 0.0
+    active_stream: None or Stream = None
```

Comment:
This is not the syntax for defining this type.
It should be either `Optional[Stream]` or `Stream | None`
Note the later syntax is only available in python >= 3.10 (afaik)